StudentDetails.java

```diff
@@ -0,0 +1,155 @@
+                    System.out.printf("%-10s %-15s %-15s", "id", "firstName", "LastName");
+                    System.out.println();
+                    Iterator<StudentData> i = c.iterator();
+                    while (i.hasNext()) {
+                        StudentData e = i.next();
+                        System.out.println();
+                        System.out.printf("%-10s %-15s %-15s", e.getSno(), e.getSfname(), e.getSlname());
+
+                    }
+                    System.out.println();
+                    System.out.println("--------------------------------------------------");
+                    break;
+                case SEARCH_STUDENT:
+                    boolean found = false;
+                    System.out.println("Please give the student id to search? ");
+                    System.out.print("Student id:");
+                    int sno1 = obj.nextInt();
+                    System.out.println("--------------------------------------------------");
+                    i = c.iterator();
+                    while (i.hasNext()) {
+                        StudentData e = i.next();
+                        if (e.getSno() == sno1) {
+                            System.out.println(e);
+                            found = true;
+                        }
+                    }
+
+                    if (!found) {
+                        System.out.println("Record Not Found");
+                    }
+                    System.out.println("--------------------------------------------------");
+                    break;
+
+                case REMOVE_STUDENT:
+                    found = false;
+                    System.out.println("Please provide student id? ");
+                    System.out.print("Student id: ");
+                    sno1 = obj.nextInt();
+                    System.out.println("--------------------------------------------------");
+                    i = c.iterator();
+                    while (i.hasNext()) {
+                        StudentData e = i.next();
+                        if (e.getSno() == sno1) {
+                            i.remove();
+                            found = true;
+                        }
+                    }
+
+                    if (!found) {
+                        System.out.println("Record Not Found");
+                    } else {
+                        System.out.println("Record is Deleted Successfully...!");
+                    }
+
+                    System.out.println("--------------------------------------------------");
+                    break;
+                case UPDATE_STUDENT:
+                    found = false;
+                    System.out.println("Which student do you want to excute? ");
+                    System.out.print("Student id: ");
+                    sno1 = obj.nextInt();
+
+                    ListIterator<StudentData> li = c.listIterator();
```

Comment:
Instead of using an Iterator, you can also just use a normal for each loop.
for(StudentData currentStudent : c) { ... }
You can also use stream and filter like described in the Moodle page -> https://rit-moodle.inso.tuwien.ac.at/mod/page/view.php?id=662